flask_example/cmds_validate/cmds_validate.py

```diff
@@ -0,0 +1,584 @@
+
+
+@cmds_validate.route('/', methods=['GET'])
+def get_cmds_validate():
+    """
+    Service to validate kadi commands v2 against telemetry
+    """
+    try:
+        parser = reqparse.RequestParser()
+        parser.add_argument('scenario', help='scenario', type=str, default=None)
+        parser.add_argument('days', help='Lookback days', type=float, default=4)
+        parser.add_argument('run_start_time', help='Run start time', type=str, default=None)
+        args = parser.parse_args()
+        html = validate_cmd_states(**dict(args))
+    except werkzeug.exceptions.BadRequest as e:
+        return {'error': str(e), 'scenario': {}}, 400
+
+    except Exception:
+        stack_trace = traceback.format_exc()
+        msg = ('<!doctype html><meta charset=utf-8><title>shortest html5</title>'
+               + f'<pre>{stack_trace}</pre>')
+        return msg, 500
+
+    return html, 200
+
+
+# @cmds_validate.route('/', methods=['GET'])
+# def view_obspar():
+#     """
+#     View observation parameters.
+#     """
+#     parser = reqparse.RequestParser()
+#     parser.add_argument('obsid', help='ID of the observation', type=int, default=0)
+#     args = parser.parse_args()
+#     pars = cda.get_ocat_local(obsid=args.obsid)
+#     if len(pars) == 0:
+#         abort(404)
+#         # return render_template('404.html')
+
+#     return render_template('obspar.html', pars=pars)
+
+plt.rcParams['axes.formatter.limits'] = (-4, 4)
+plt.rcParams['font.size'] = 9
+TASK = 'validate_states'
+VERSION = 5
+TASK_DATA = os.path.join(os.environ['SKA'], 'data', TASK)
+URL = "http://cxc.harvard.edu/mta/ASPECT/" + TASK
+logger = logging.getLogger(TASK)
+
+TITLE = {'dp_pitch': 'Pitch',
+         'obsid': 'OBSID',
+         'tscpos': 'TSCPOS (SIM-Z)',
+         'pcad_mode': 'PCAD MODE',
+         'dither': 'DITHER',
+         'letg': 'LETG',
+         'hetg': 'HETG',
+         'pointing': 'Commanded ATT Radial Offset',
+         'roll': 'Commanded ATT Roll Offset'}
+
+LABELS = {'dp_pitch': 'Pitch (degrees)',
+          'obsid': 'OBSID',
+          'tscpos': 'SIM-Z (steps/1000)',
+          'pcad_mode': 'PCAD MODE',
+          'dither': 'Dither',
+          'letg': 'LETG',
+          'hetg': 'HETG',
+          'pointing': 'Radial Offset (arcsec)',
+          'roll': 'Roll Offset (arcsec)'}
+
+
+SCALES = {'tscpos': 1000.,
+          'dither': 1.}
+
+FMTS = {'dp_pitch': '%.3f',
+        'obsid': '%d',
+        'dither': '%d',
+        'hetg': '%d',
+        'letg': '%d',
+        'pcad_mode': '%d',
+        'tscpos': '%d',
+        'pointing': '%.2f',
+        'roll': '%.2f',
+        }
+
+MODE_SOURCE = {'pcad_mode': 'aopcadmd',
+               'dither': 'aodithen'}
+
+MODE_MSIDS = {'pcad_mode': ['NMAN', 'NPNT', 'NSUN', 'NULL',
+                            'PWRF', 'RMAN', 'STBY'],
+              'dither': ['ENAB', 'DISA'],
+              'hetg': ['INSE', 'RETR'],
+              'letg': ['INSE', 'RETR'],
+              }
+
+
+def config_logging(outdir, verbose):
```

Comment:
Normally one would not have logging configuration inside a blueprint. It would be done at the app level instead.